test/clj_diff/test/custom_comparator.clj

```diff
@@ -0,0 +1,71 @@
+(ns clj-diff.test.custom-comparator
+  (:use [clj-diff.core]
+        [clojure.test]))
+
+(defn ignore-case [a b]
+  (cond
+    (and (char? a) (char? b)) (= (Character/toLowerCase ^char a) (Character/toLowerCase ^char b))
+    (and (string? a) (string? b)) (= (.toLowerCase ^String a) (.toLowerCase ^String b))
+    :else (= a b)))
+
+(deftest diff-test
+  (let [t (fn [a b] (edit-distance (diff a b {:compare ignore-case})))]
+
+    (is (= (edit-distance (diff "aaaabbcc" "AAAAbBcC"))
+          12))
+    (is (= (edit-distance (diff "aaaabbcc" "AAAAbBcCz"))
+          13))
+    (is (= (edit-distance (diff "aaaabbcc" "AAAAbBcC" {:compare ignore-case}))
+          0))
+    (is (= (edit-distance (diff "aaaabbcc" "AAAAbBcCz" {:compare ignore-case}))
+          1))))
+
+(deftest patch-test
+  (is (= (patch "aaaabbcc"
+           (diff "aaaabbcc" "AAAAbBcC" {:compare ignore-case}))
+        "aaaabbcc"))
+
+  (is (= (patch "aaaabbcc"
+           (diff "aaaabbcc" "AAAAbBcCz" {:compare ignore-case}))
+        "aaaabbccz"))
+
+  (is (= (patch "aaaabbcc"
+           (diff "aaaabbcc" "AAAAbBcCZ" {:compare ignore-case}))
+        "aaaabbccZ")))
+
+
+(deftest edit-distance-test
+  (is (= (edit-distance (diff "aba" "aBa"))
+        2))
+  (is (= (edit-distance (diff "aba" "aBa" {:compare ignore-case}))
+         0))
+  (is (= (edit-distance (diff "aBcaBBa" "cbabac"))
+        7))
+  (is (= (edit-distance (diff "aBcaBBa" "cbabac" {:compare ignore-case}))
+         5))
+  (is (= (edit-distance (diff "nBP8GaFHVls2dI8h9aK1FWdRgevf43"
+                              "925BCPcYhT5hs8L9T3K2T5C7U3Lz5v"
+                          {:compare ignore-case}))
+         44)))
+
+(deftest longest-common-subseq-test
```

Comment:
I'd like it if this namespace were integrated into `clj-diff.test.core`. Particularly because a few of these tests are just slight modifications of tests there e.g. https://github.com/droitfintech/clj-diff/blob/88a5e52bc7786dfd7485ca5594b63a365e82c391/test/clj_diff/test/core.clj#L118-L128 

In a separate namespace that doesn't actually correspond to one in prod, these tests aren't particularly discoverable.

I won't block on this though as this is capturing important functionality but I'd like it cleaned up in a separate PR before this leaves alpha.